dist/api_err.py

```diff
@@ -34,7 +34,9 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
         This error is generated when an operation cannot be completed
         due to a conflict with concurrent operations.  The operation
         may be retried; if a transaction is in progress, it should be
-        rolled back and the operation retried in a new transaction.'''),
+        rolled back and the operation retried in a new transaction.
+        WiredTiger reserves the right to return WT_ROLLBACK from any
+        cursor API call.'''),
```

Comment:
I suppose, WiredTiger reserves the right to return any error on any API call unless stated otherwise. Is there anything special about `WT_ROLLBACK` that it's worth mentioning it explicitly?

Reply:
I think the reason we mention it specifically here is that it prevents further questions about cursor APIs and rollback. Though I see your point, if we mention that here then it raises further questions about what error codes are valid elsewhere. I'll see if I can document this.

Reply:
Reworded this in: e7037c21b